### ISSUE TITLE
test suite is broken on node master

### ISSUE THREAD

**Issue by TheAlphaNerd** [eslint/eslint#5678]
**What version of ESLint are you using?**
Master

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
testing with current head of node.js master on OSX 10.10.5

**What did you do? Please include the actual source code causing the issue.**
```
$ npm install
$ npm test
```

**What did you expect to happen?**
The test suite to pass

**What actually happened? Please include the actual, raw output from ESLint.**
```
 9920 passing (44s)
  6 failing

  1) CLIEngine executeOnText() should report one message when using local cwd .eslintrc:
     TypeError: Path must be a string. Received undefined
      at assertPath (path.js:7:11)
      at Object.extname (path.js:1469:5)
      at processText (lib/cli-engine.js:9:2174)
      at CLIEngine.executeOnText (lib/cli-engine.js:9:19381)
      at Context.<anonymous> (tests/lib/cli-engine.js:91:33)

  2) CLIEngine executeOnText() should report one message when using specific config file:
     TypeError: Path must be a string. Received undefined
      at assertPath (path.js:7:11)
      at Object.extname (path.js:1469:5)
      at processText (lib/cli-engine.js:9:2174)
      at CLIEngine.executeOnText (lib/cli-engine.js:9:19381)
      at Context.<anonymous> (tests/lib/cli-engine.js:110:33)

  3) CLIEngine getErrorResults() should report 4 error message when looking for errors only:
     TypeError: Path must be a string. Received undefined
      at assertPath (path.js:7:11)
      at Object.extname (path.js:1469:5)
      at processText (lib/cli-engine.js:9:2174)
      at CLIEngine.executeOnText (lib/cli-engine.js:9:19381)
      at Context.<anonymous> (tests/lib/cli-engine.js:1871:33)

  4) CLIEngine when evaluating code with comments to change config when allowInlineConfig is disabled should report a violation for disabling rules:
     TypeError: Path must be a string. Received undefined
      at assertPath (path.js:7:11)
      at Object.extname (path.js:1469:5)
      at processText (lib/cli-engine.js:9:2174)
      at CLIEngine.executeOnText (lib/cli-engine.js:9:19381)
      at Context.<anonymous> (tests/lib/cli-engine.js:2013:36)

  5) CLIEngine when evaluating code with comments to change config when allowInlineConfig is disabled should not report a violation by default:
     TypeError: Path must be a string. Received undefined
      at assertPath (path.js:7:11)
      at Object.extname (path.js:1469:5)
      at processText (lib/cli-engine.js:9:2174)
      at CLIEngine.executeOnText (lib/cli-engine.js:9:19381)
      at Context.<anonymous> (tests/lib/cli-engine.js:2039:36)

  6) cli execute() should return error when text with incorrect quotes is passed as argument:
     TypeError: Path must be a string. Received undefined
      at assertPath (path.js:7:11)
      at Object.extname (path.js:1469:5)
      at processText (lib/cli-engine.js:9:2174)
      at CLIEngine.executeOnText (lib/cli-engine.js:9:19381)
      at Object.cli.execute (lib/cli.js:9:5669)
      at Context.<anonymous> (tests/lib/cli.js:115:30)
```

**Comment by nzakas** [comment]
Hmm, how long ago did you pull from master? I *just* did a release using master and the tests passed fine during that.

**Comment by nzakas** [comment]
Oh I see, you're using Node.js master. Shouldn't file an issue with Node.js for this problem?

**Comment by TheAlphaNerd** [comment]
I am one of the collaborators on the node project. This was discovered during smoke testing. I wanted to let ya'll know that you are broken on master as we will be cutting v6 in the next month. Some of the major changes on master are breaking. I'm doing some forensics to figure out exactly what it is.

I'm not 100% if this is a regression in node yet, or a change will need to be made in eslint due to Semver Major changes.

**Comment by TheAlphaNerd** [comment]
@nzakas just finished bisecting... and it looks like this is the offending commit
https://github.com/nodejs/node/commit/08085c49b6e49ecfffd638c6cd46f4de639ae4f4

it is a semver major change to assert that the values given to path are strings. I'm going to quickly dig into the tests on your end and see if this can be patched in eslint easily.
